Makefile

```diff
@@ -107,10 +108,20 @@ images: $(foreach dir,$(IMAGE_DIRS),image-$(dir))
 # Rules for building and installing binaries, or building docker images, and cleaning up.
 #
 
+KERNEL_INCLUDE_DIRS = /include \
+                      /include/uapi \
+                      /include/generated/uapi \
+                      /arch/x86/include \
+                      /arch/x86/include/uapi \
+                      /arch/x86/include/generated/uapi
+
+KERNEL_INCLUDES := $(foreach kernel_dir,$(KERNEL_SRC_DIR) $(KERNEL_BUILD_DIR),$(addprefix -I,$(wildcard $(addprefix $(kernel_dir),$(KERNEL_INCLUDE_DIRS)))))
+
 libexec/%.o: elf/%.c
+	@if [ -z $(KERNEL_INCLUDES) ]; then echo "Cannot build $@: invalid KERNEL_SRC_DIR=$(KERNEL_SRC_DIR)"; exit 1; fi
```

Comment:
Oh, and I think this is wrong... it should be if [ -z "$(KERNEL_INCLUDES)" ]; ... to avoid shell evaluation errors.

Reply:
Good findings.
- Added missing doublequotes.
- Added `$(strip ... )` around `KERNEL_INCLUDES`, because otherwise this `if` never prints user-friendly error. Without trimming `KERNEL_INCLUDES` it seems to always include one space.
- Added `$(Q)`.